File: frontend/src/questionnaires/dto/QuestionnaireResponseEditorDto.ts
import {QuestionnaireStatus} from "./QuestionnaireStatus.ts";
import {UserResponsePublicDto} from "../../user/dto/UserResponsePublicDto.ts";

export interface QuestionnaireResponseEditorDto {
  id: number;
  name: string;
  description: string;
  status: QuestionnaireStatus;
  maxPoints: number;
  createdBy: UserResponsePublicDto;
  createdAt: Date;
  updatedBy: UserResponsePublicDto;
  updatedAt: Date;
}

File: backend/src/main/java/net/dnadas/training_portal/dto/group/project/questionnaire/QuestionnaireResponseEditorDto.java
package net.dnadas.training_portal.dto.group.project.questionnaire;

import net.dnadas.training_portal.dto.user.UserResponsePublicDto;
import net.dnadas.training_portal.model.group.project.questionnaire.QuestionnaireStatus;

public record QuestionnaireResponseEditorDto(
  Long id, String name, String description,
  QuestionnaireStatus status,
  Integer maxPoints,
  UserResponsePublicDto createdBy, String createdAt,
  UserResponsePublicDto updatedBy, String updatedAt) {
}
